scripts/fix-engine-names.py

```diff
@@ -64,17 +64,84 @@ def addSufiTaggedEngines():
         exceptionalNames[name2] = i
 
 
+def addFixedStockfish15():
+    # change "Stockfish_15_100M" to "Stockfish_100M (15)" etc.
+    for i in [1, 3, 10, 30, 100, 300]:
+        for s in ["k", "M", "G"]:
+            suffix = f"{i}{s}"
+            name = "Stockfish_15_" + suffix
+            newname = "Stockfish_" + suffix
+            exceptionalNames[name] = (newname, "15", "")
+    exceptionalNames["Stockfish_15_300M scaled_to_450Mnodes"] = (
+        "Stockfish_300M",
+        "15",
+        "scaled to 450Mnodes",
+    )
+    exceptionalNames["Renamedfish_15_3M"] = (
+        "Stockfish_3M",
+        "15",
+        "renamed",  # or "copy"
+    )
+
+
+def amendEngineNames(player):
```

Comment:
I would rather have all these added in the exceptional names dictionary, so that every engine name substitution is spelled out in the dictionary.

Two reasons:

1. Just by printing out the dictionary, one can inspect the substitutions exactly one by one
2. It's too easy for substitution lists like these to have unintended consequences.3.